applications/nexus/nex2ecl.cpp

```diff
@@ -16,37 +16,148 @@
+                             output at <path>.
+)";
+
+static struct option long_options[] = {
+    {"format", no_argument,       nullptr, 'f'},
+    {"help",   no_argument,       nullptr, 'h'},
+    {"output", required_argument, nullptr, 'o'}
+};
 
-void create_ecl_sum(char *file_path) {
-    std::string filename(file_path);
-    auto ending = filename.rfind(".plt");
+}
+
+void create_ecl_sum( std::string input_path,
+                     std::string output_path,
+                     bool format_output ) {
+    auto ending = input_path.rfind(".plt");
     if (ending == std::string::npos) {
-        filename += ".plt";
+        input_path += ".plt";
     }
 
-    nex::NexusPlot plt = nex::load(filename);
+    std::string ecl_case_name {};
+    {
+        char *plt_basename;
+        util_alloc_file_components( input_path.c_str(),
+                                    NULL,
+                                    &plt_basename,
+                                    NULL );
+        util_strupr( plt_basename );
+
+        if ( output_path.empty() ) {
+            ecl_case_name = std::string( plt_basename );
+        } else if ( util_is_directory(output_path.c_str()) ) {
+            ecl_case_name = output_path + std::string( plt_basename );
+        } else {
+            char *path = 0, *basename = 0;
+            util_alloc_file_components(output_path.c_str(), &path, &basename, NULL);
+
+            if ( !basename ) {
+                std::cerr << "Failed to parse output path "
+                          << output_path << "." << std::endl;
+                exit(1);
+            }
+
+            if ( path && !util_is_directory(path) ) {
+                std::cerr << path << " does not exists." << std::endl;
+                exit(1);
+            }
+
+            util_strupr( basename );
 
-    char * basename;
-    util_alloc_file_components( file_path , NULL, &basename , NULL);
-    util_strupr( basename );
+            if ( path ) {
+                ecl_case_name += std::string( path );
+                ecl_case_name += UTIL_PATH_SEP_CHAR;
+            }
+            ecl_case_name += std::string( basename );
 
-    ecl_sum_type *ecl_sum = nex::ecl_summary(basename, plt);
+            if ( path ) free( path );
+            if ( basename ) free( basename );
+        }
+
+        free( plt_basename );
+    }
+
+    nex::NexusPlot plt = nex::load( input_path );
+    ecl_sum_type *ecl_sum = nex::ecl_summary( ecl_case_name.c_str(),
+                                              format_output,
+                                              plt );
     ecl_sum_fwrite(ecl_sum);
     ecl_sum_free(ecl_sum);
-    free( basename );
 }
 
+int main( int argc, char **argv ) {
+    if (argc == 1) {
+        std::cout << usage << std::endl;
+        exit(1);
+    }
+
+    bool format_output = false;
+    std::string output_path {};
+    std::string input_path {};
+
+    /*
+     * Parse options
+     */
+
+    int c;
```

Comment:
No reason to declare `c`outside the `for` loop?

Reply:
No, there isn't, fixing